test/custom_console_dashboard_test.go

```diff
@@ -0,0 +1,108 @@
+//go:build e2e
+// +build e2e
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/go-github/v50/github"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params"
+	tgithub "github.com/openshift-pipelines/pipelines-as-code/test/pkg/github"
+	twait "github.com/openshift-pipelines/pipelines-as-code/test/pkg/wait"
+	"gotest.tools/v3/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	customConsoleDashboardName        = "custom-console-name"
+	customConsoleDashboardURL         = "custom-console-url"
+	customConsoleDashboardPRDetailURL = "custom-console-url-pr-details"
+	customConsoleDashboardPRTaskLog   = "custom-console-url-pr-tasklog"
+)
+
+func TestCustomConsoleDashboard(t *testing.T) {
+	ctx := context.TODO()
+	runcnx, ghcnx, opts, targetNS, targetRefName, prNumber, sha := tgithub.RunPullRequest(ctx, t,
+		"Github Retest comment", []string{"testdata/pipelinerun.yaml"}, false)
+	defer tgithub.TearDown(ctx, t, runcnx, ghcnx, prNumber, targetRefName, targetNS, opts)
+
+	prList, err := runcnx.Clients.Tekton.TektonV1().PipelineRuns(targetNS).List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("pipelinesascode.tekton.dev/sha=%s", sha),
+	})
+	assert.NilError(t, err)
+	assert.Assert(t, strings.Contains(prList.Items[0].Annotations["pipelinesascode.tekton.dev/log-url"], "https://console-openshift-console.apps"))
+	// update config map with console dashboard information
+	cmList, er := runcnx.Clients.Kube.CoreV1().ConfigMaps("").List(ctx, metav1.ListOptions{})
+	assert.NilError(t, er)
+	for _, v := range cmList.Items {
+		if v.Name == "pipelines-as-code" {
+			cmData, err := runcnx.Clients.Kube.CoreV1().ConfigMaps(v.Namespace).Get(ctx, v.Name, metav1.GetOptions{})
+			assert.NilError(t, err)
+			cmData.Data[customConsoleDashboardName] = "My custom console"
```

Comment:
I haven't done the e2e test for the configmap setting because when you modify globally setting like that it may fail on a different test, which check for log url if it's properly reported...

hopefully since we are running sequentially this won't break since you are reverting after that.

and generally it's better for this of test to do this in gitea, since the github one gets rate limited quickly when we have one...

Reply:
Yes 
Sure
Changed Github tests to Gitea :+1:

Reply:
nice! can you prefix with TestGitea, that's how we can scope the test to the right provider?